ruby-challenges.rb

```diff
@@ -10,18 +10,57 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+#create a method named includes_letter that uses .include? to check for the specific letters
+#also use .filter to return the wanted strings and leave out the unwanted strings in a new array
+
+   
+  def includes_letter(givenArray, string)
+    givenArray.filter {|newArray| newArray.include? string}
+  end
+
+  p includes_letter(beverages_array, letter_o)
+  p includes_letter(beverages_array, letter_t)
+
+  #output: ["coffee", "soda water"]
+  #output: ["tea", "water", "soda water"]
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
+#from the hint, I found that .flatten takes all of the elements from multiple arrays and combines them into a new array, after that use .sort to get all of the states in alphabetical order but before all of that you have to use .values to access the values then you can apply .flatten and .sort
+
+p us_states.values.flatten.sort
+
+#output: ["California", "Arizona", "Nevada", "Maine", "New Hampshire", "Vermont", "Washington", "Oregon", "Idaho"]
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+# make a class named Bike, but include the initialize method that will run everytime .new is called upon
+# have a getter method to get bike info and a setter method to assign values
+# add an attribute accessor to update the model in this case, but you can also update the speed and the number of wheels if needed
+#use string interpolation in order for the output to display the model, wheel, and speed
+
+class Bike 
+    attr_accessor :model, :wheels, :current_speed
+    def initialize(model)
+      @model = model
+      @wheels = 2
+      @current_speed = 0
+    end
+    def bike_info
+      "The #{@model} bike has #{@wheels} wheels, and is going #{@current_speed}mph."
+    end
+  
+  p my_bike = Bike.new('Trek') 
+  p my_bike.bike_info 
```

Comment:
It looks like this is living inside of the class currently.  When creating a new instance or calling on a method, that will take place outside the class.